### DISCUSSION TITLE
target:installdir("xx") 不会自动创建文件夹

### DISCUSSION THREAD

**Discussion by TOMO-CAT** [xmake-io/xmake#4674] [Q&A]
### Xmake 版本

2.8.5+20231124

### 操作系统版本和架构

Linux 720ce3a659a2 5.15.90.1-microsoft-standard-WSL2 #1 SMP Fri Jan 27 02:56:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### 描述问题

package:installdir() 会自动创建不存在的目录树，target:installdir() 没有这个特性，是遗漏了还是出于其他考量呢？
![image](https://github.com/xmake-io/xmake/assets/47598093/5b56a9ca-1127-4213-a3f7-df6c8a5c2334)

### 期待的结果

希望 target:installdir 也提供这个功能

### 工程配置

无

### 附加信息和错误日志

无

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
> package:installdir() 会自动创建不存在的目录树，target:installdir() 没有这个特性，是遗漏了还是出于其他考量呢？

package:installdir() 用于包的安装，自动创建目录只是为了简化包的 on_install 脚本，反正安装包总共要创建目录的，其他地方也不会访问到这个接口。。


target:intalldir() 在任何脚本中都能访问，不一定就是在安装过程中，不会去默认创建它。。也可能大部分场景仅仅只是 readonly，去创建反而有问题
